epictrack-api/src/api/resources/project.py

```diff
@@ -1,189 +1,205 @@
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def get(project_id):
+        """Return work types associated with a project."""
+        req.ProjectIdPathParameterSchema().load(request.view_args)
+        args = req.WorkIdPathParameterSchema().load(request.args)
+        work_id = args["work_id"]
+        work_types = ProjectService.find_project_work_types(project_id, work_id)
+        return WorkTypeSchema(many=True).dump(work_types), HTTPStatus.OK
+
+
+@cors_preflight("GET")
+@API.route("/<int:project_id>/first-nations", methods=["GET", "OPTIONS"])
+class ProjectFirstNations(Resource):
+    """Endpoint resource to get all first nations associated with a project."""
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def get(project_id):
+        """Return first nations associated with a project."""
+        req.ProjectIdPathParameterSchema().load(request.view_args)
+        args = req.ProjectFirstNationsQueryParamSchema().load(request.args)
+        work_type_id = args["work_type_id"]
+        work_id = args["work_id"]
+        first_nations = ProjectService.find_first_nations(
+            project_id, work_id, work_type_id
+        )
+        return (
+            res.IndigenousResponseNationSchema(many=True).dump(first_nations),
+            HTTPStatus.OK,
+        )
+
+
+@cors_preflight("GET")
+@API.route("/<int:project_id>/first-nation-available", methods=["GET", "OPTIONS"])
+class ProjectFirstNationAvailableStatus(Resource):
+    """Endpoint resource to check if there are first nations associated with a project."""
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def get(project_id):
+        """Check if any first nations associated with given project."""
+        req.ProjectIdPathParameterSchema().load(request.view_args)
+        args = req.ProjectFirstNationsQueryParamSchema().load(request.args)
+        work_id = args["work_id"]
+        first_nation_availability = ProjectService.check_first_nation_available(
+            project_id, work_id
+        )
+        return first_nation_availability, HTTPStatus.OK
+
+
+@cors_preflight("POST")
+@API.route("/import", methods=["POST", "OPTIONS"])
+class ImportProjects(Resource):
+    """Endpoint resource to import projects."""
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def post():
+        """Import projects"""
+        file = request.files["file"]
+        response = ProjectService.import_projects(file)
+        return response, HTTPStatus.CREATED
+
+
+@cors_preflight("POST")
+@API.route("/abbreviation", methods=["POST", "OPTIONS"])
+class ProjectAbbreviation(Resource):
+    """Endpoint resource to import projects."""
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def post():
+        """Create new project abbreviation"""
+        request_json = request.get_json()

```

Comment:
Can we use a schema here? That way we can add validations for required and min lengths and avoid defaulting to `''` in the function call below.

Reply:
I can add a schema, I'm not sure whether I should add a minimum check. The abbreviation could be custom, which means it can work with any name.